pkg/azuredisk/nodeserver.go

```diff
@@ -324,6 +324,11 @@ func (d *Driver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (
 		Segments: map[string]string{topologyKey: ""},
 	}
 
+	failureDomain, instanceType, err := getNodeInfoFromAPIServer(ctx, d.NodeID, d.cloud.KubeClient)
```

Comment:
this impl is always trying to get the API server, in the patch I sent I think it was behind a flag

Reply:
what about this? if get failureDomain and instanceType failed from instance metadata, then we always fall back to get from apiserver? use a flag to control this behavior is difficult to use (always need manual config). And even get from instance metadata timeout, it would proceed to get from apiserver.

Reply:
> if get failureDomain and instanceType failed from instance metadata

If it's not available then the call will block for some time before it times out and continues, it'd be great if we could instead continue immediately if we detected that the metadata server is unavailable, my concern is with the time NodeGetInfo takes before it succeeds

Reply:
sure, I will change the logic a little, and also I found this requires nodes get role: https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1200/commits/f2014c514fb7e3e6952d8502b1459145a38a39de